### PR TITLE
MOBT-5-UI-From code-with-Core-Data

### PR DESCRIPTION
**Pull request by MykytaBabanin**
MOBT-5-UI-From code-with-Core-Data
In scope of this task was developed: 
Main screen with different chats.
Screen with individual user, that have messages from Core Data
Tab bar which is similar with telegram.
image
image

### CODE REVIEW

File: MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Friends/FriendsControllerHelper.swift [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                                   context: context)
+            createMessage(with: "fsafsafasfasfasfsafsafasfsafsafsafsa", friend: steve, minutesAgo: 20, context: context)
+            createMessage(with: "afsfsafasfsafsafsafsafasfsa", friend: mark, minutesAgo: 10, context: context)
+            createMessage(with: "Hello, my name is Gandhidsadsadsadsadsadsadsadsadsadsadsadsadsadsadas", friend: gandhi, minutesAgo: 5, context: context)
+            do {
+                try (context.save())
+            } catch let err {
+                print(err)
+            }
+            messages = messages?.sorted(by: {$0.date!.compare($1.date!) == .orderedDescending })
```
Comment (kverkhoturov) [inline MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Friends/FriendsControllerHelper.swift:50]: force unwrap

File: MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Chat/ChatLogMessageCell.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+//  MOBT-5-Dialog-Messenger
+//
+//  Created by Mykyta Babanin on 20.04.2022.
+//
+
+import UIKit
+
+class ChatLogMessageCell: BaseCell {
+    
+    lazy var messageTextView = UITextView()
```
Comment (kverkhoturov) [inline MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Chat/ChatLogMessageCell.swift:12]: must be private

File: MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Friends/FriendsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+//  Created by Mykyta Babanin on 10.04.2022.
+//
+
+import UIKit
+
+class FriendsViewController: UICollectionViewController, UICollectionViewDelegateFlowLayout {
+    
+    private let cellId = "cellId"
+    
+    var messages: [Message]?
```
Comment (kverkhoturov) [inline MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Friends/FriendsViewController.swift:14]: massive view controller

File: MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Chat/ChatLogController.swift [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+//  Created by Mykyta Babanin on 12.04.2022.
+//
+
+import UIKit
+
+class ChatLogController: UICollectionViewController, UICollectionViewDelegateFlowLayout {
+    
+    private let cellId = "cellId"
+    
+    var friend: Friend? {
```
Comment (kverkhoturov) [inline MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Chat/ChatLogController.swift:14]: massive view controller

File: MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Friends/FriendsControllerHelper.swift [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        loadData()
+    }
+    
+    private func createUser(entityName: String,
+                            userName: String,
+                            profileImage: String,
+                            chatMessage: String,
+                            minutesAgo: Double,
+                            context: NSManagedObjectContext) -> Friend {
+        let user = NSEntityDescription.insertNewObject(forEntityName: entityName, into: context) as! Friend
```
Comment (kverkhoturov) [inline MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Friends/FriendsControllerHelper.swift:61]: force unwrap

File: MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Friends/FriendsControllerHelper.swift [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    func loadData() {
+        let delegate = UIApplication.shared.delegate as? AppDelegate
+        if let context = delegate?.persistentContainer.viewContext {
+            
+            if let friends = fetchFriends() {
+                messages = [Message]()
+                for friend in friends {
+                    let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: Consts.entityMessageName)
+                    fetchRequest.sortDescriptors = [NSSortDescriptor(key: "date", ascending: false)]
+                    fetchRequest.predicate = NSPredicate(format: "friend.name = %@", friend.name!)
```
Comment (kverkhoturov) [inline MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Friends/FriendsControllerHelper.swift:84]: force unwrap

File: MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Friends/FriendsControllerHelper.swift [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+    
+    private func clearData() {
+        let delegate = UIApplication.shared.delegate as? AppDelegate
+        if let context = delegate?.persistentContainer.viewContext {
+            do {
+                let entityNames = [Consts.entityFriendName, Consts.entityMessageName]
+                for entityName in entityNames {
+                    let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: entityName)
+                    let objects = try(context.fetch(fetchRequest)) as? [NSManagedObject]
+                    for object in objects! {
```
Comment (kverkhoturov) [inline MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Friends/FriendsControllerHelper.swift:118]: force unwrap

File: MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Chat/ChatLogMessageCell.swift (hunk 2 above)
Comment (MykytaBabanin) [reply MOBT-5-Dialog-Messenger/MOBT-5-Dialog-Messenger/Chat/ChatLogMessageCell.swift:12]: messageTextView was used in chatLogController when I dequeued cell.

**Review by kverkhoturov — APPROVED** [review]

**Pull request merged**